src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -760,20 +760,11 @@ public static FormulaValue ModFloat(IRContext irContext, NumberValue arg0nv, Num
                 return CommonErrors.DivByZeroError(irContext);
             }
 
-            // r = a – N × floor(a/b)
-            double q = Math.Floor(arg0 / arg1);
-            if (IsInvalidDouble(q))
-            {
-                return CommonErrors.OverflowError(irContext);
-            }
-
-            double result = arg0 - (arg1 * ((long)q));
+            double result = arg0 % arg1;
 
-            // We validate the reminder is in a valid range.
-            // This is mainly to support very large numbers (like 1E+308) where the calculation could be incorrect
-            if (result < -Math.Abs(arg1) || result > Math.Abs(arg1))
+            if (Math.Sign(result) * Math.Sign(arg1) == -1)
             {

```

Comment:
We should add a comment to explain what we are doing here

Reply:
Added comment. thanks